### PR TITLE
Fixes #1753: Redux integration for Forgot password component

### LINKED ISSUE #1753: Integrateing redux in ForgotPassoword.react.js
**Actual Behaviour**

ForgotPassword.react.js has not been integrated with the Redux store.

**Expected Behaviour**

It must be integrated into the new redux implementation

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1753 

Changes: The changes I made are:
- [x] Grouping imports in a logical fashion
- [x] Renaming variables so as to reduce the need for comments
- [x] Make functions simpler wherever possible
- [x] Put styles out of the component
- [x] Convert normal functions to arrow functions
- [x] Move propType validation inside class component.
- [x] Integrated redux for this component. 

Demo Link: https://pr-1755-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

File: src/components/Auth/ForgotPassword/ForgotPassword.react.js [hunk 1]
```diff
@@ -1,234 +1,132 @@
...
+    height: '26px',
+    right: '10px',
+    top: '10px',
+    cursor: 'pointer',
+  },
+};
 
 class ForgotPassword extends Component {
+  static propTypes = {
+    onLoginSignUp: PropTypes.func,
```
Comment (akshatnitd) [inline src/components/Auth/ForgotPassword/ForgotPassword.react.js]: Is this prop being used ?

File: src/components/Auth/ForgotPassword/ForgotPassword.react.js [hunk 2]
```diff
@@ -1,234 +1,132 @@
...
-          state.loading = false;
-          this.setState(state);
-        }.bind(this),
-        error: function(jqXHR, textStatus, errorThrown) {
-          let jsonValue = jqXHR.status;
-          let msg = '';
-          if (jsonValue === 404) {
-            msg = 'Email does not exist';
+    if (email && !emailErrorMessage) {
+      this.setState({ loading: true });
```
Comment (akshatnitd) [inline src/components/Auth/ForgotPassword/ForgotPassword.react.js]: Not required.

File: src/components/Auth/ForgotPassword/ForgotPassword.react.js [hunk 3]
```diff
@@ -1,234 +1,132 @@
...
+            loading: false,
+          });
+        })
+        .catch(error => {
+          this.setState({
+            loading: false,
+            dialogMessage: 'Failed. Try Again',
+          });
+        });
     }
```
Comment (akshatnitd) [inline src/components/Auth/ForgotPassword/ForgotPassword.react.js]: Make loading false here

File: src/components/Auth/ForgotPassword/ForgotPassword.react.js [hunk 4]
```diff
@@ -1,234 +1,132 @@
...
 import RaisedButton from 'material-ui/RaisedButton';
 import Dialog from 'material-ui/Dialog';
 import Close from 'material-ui/svg-icons/navigation/close';
-
-/* CSS*/
-import './ForgotPassword.css';
-
-/* Utils*/
-import $ from 'jquery';
 import UserPreferencesStore from '../../../stores/UserPreferencesStore';
```
Comment (akshatnitd) [inline src/components/Auth/ForgotPassword/ForgotPassword.react.js]: Is it required now?

Comment (AakashMallik) [reply src/components/Auth/ForgotPassword/ForgotPassword.react.js]: User preference is being used to get themes colors in render function for now. If it had not been used, Eslint would have pointed it out.

File: src/components/Auth/ForgotPassword/ForgotPassword.react.js (hunk 3 above)
Comment (AakashMallik) [reply src/components/Auth/ForgotPassword/ForgotPassword.react.js]: I have shifted {loading: true} inside the if block. So I guess {loading: false} outside if block will not be required any more

**Comment by AakashMallik** [comment]
@akshatnitd Sorry sir. I had my focus on integrating redux so I guess I missed those minor defects in code. I have made the changes you asked for and have proposed some counter-proposal for a few more. Let me know what you think. Thank you for your time.

File: src/redux/actionTypes/index.js [hunk 5]
```diff
@@ -1,3 +1,7 @@
 import _ from 'lodash';
 
-export default _.mapKeys(['APP_GET_LOGIN', 'APP_GET_API_KEYS']);
+export default _.mapKeys([
+  'APP_GET_FORGOT_PASSOWORD',
```
[the remark this answers is not in the document]
Comment (AakashMallik) [reply src/redux/actionTypes/index.js]: I feel embarrassed for this typo.

**Comment by AakashMallik** [comment]
@akshatnitd Removed the typo and unused reducer.

**Comment by AakashMallik** [comment]
ezgif-1-3f47fcbe70c8

@akshatnitd I have added a loader and also got rid of the pop up for error messages. I hope you like it.

**Comment by AakashMallik** [comment]
@akshatnitd Sir, putting the loaded as a child component of material ui button did not go well so I removed it. Rest of the things have been implemented just the way you asked. And I have changed the deploy link, it was outdated.  Thanks for your time.

**Comment by AakashMallik** [comment]
@akshatnitd Done :innocent: Changed the with of the reset button to make it look prominent. Fixed the alignment as you asked.

**Pull request merged**